internal/adapters/objectsetphase.go

```diff
@@ -0,0 +1,150 @@
+package adapters
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+)
+
+type ObjectSetPhaseAccessor interface {
+	ClientObject() client.Object
+	GetConditions() *[]metav1.Condition
+	GetClass() string
+	GetPrevious() []corev1alpha1.PreviousRevisionReference
+	GetPhase() corev1alpha1.ObjectSetTemplatePhase
+	GetAvailabilityProbes() []corev1alpha1.ObjectSetProbe
+	GetRevision() int64
+	GetGeneration() int64
+	IsSpecPaused() bool
+	SetStatusControllerOf([]corev1alpha1.ControlledObjectReference)
+}
+
+var (
+	_ ObjectSetPhaseAccessor = (*ObjectSetPhase)(nil)
+	_ ObjectSetPhaseAccessor = (*ClusterObjectSetPhase)(nil)
+)
+
+type ObjectSetPhaseFactory func(
+	scheme *runtime.Scheme) ObjectSetPhaseAccessor
+
+var (
+	objectSetPhaseGVK        = corev1alpha1.GroupVersion.WithKind("ObjectSetPhase")
+	clusterObjectSetPhaseGVK = corev1alpha1.GroupVersion.WithKind("ClusterObjectSetPhase")
+)
+
+func NewObjectSetPhaseAccessor(scheme *runtime.Scheme) ObjectSetPhaseAccessor {
+	obj, err := scheme.New(objectSetPhaseGVK)
+	if err != nil {
+		panic(err)
+	}
+
+	return &ObjectSetPhase{
+		ObjectSetPhase: *obj.(*corev1alpha1.ObjectSetPhase),
+	}
+}
+
+func NewClusterObjectSetAccessor(scheme *runtime.Scheme) ObjectSetPhaseAccessor {
+	obj, err := scheme.New(clusterObjectSetPhaseGVK)
+	if err != nil {
+		panic(err)
+	}
+
+	return &ClusterObjectSetPhase{
+		ClusterObjectSetPhase: *obj.(*corev1alpha1.ClusterObjectSetPhase),
+	}
+}
+
+type ObjectSetPhase struct {
```

Comment:
Please rename this struct (and the cluset-scoped one) so that it is clear that this is an adapter, for example: `ObjectSetPhaseAdapter`.